code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -477,30 +477,31 @@
 			if(collectItems(get_turf(A), user))
 				return TRUE
 	//Clicking on tile with no collectible items will empty it, if it has the verb to do that.
-	if(src.verbs.Find(/obj/item/weapon/storage/verb/quick_empty))
-		if(isturf(A))
-			src.quick_empty(A)
+	if(allow_quick_empty)
+		if(isturf(A) && !A.density)
+			dump_it(A)
 			return TRUE
-
 	return ..()
 
-/obj/item/weapon/storage/verb/quick_empty(var/turf/target)
+/obj/item/weapon/storage/verb/quick_empty()
 	set name = "Empty Contents"
 	set category = "Object"
+	set src in view(1)
 
 	if((!ishuman(usr) && (src.loc != usr)) || usr.stat || usr.restrained())
 		return
 
-	var/turf/T
-	if(isturf(target))
-		T = target
-	else
-		T = get_turf(src)
+	var/turf/T = get_turf(src)
 	if(!istype(T))
 		return
+	dump_it(T)
+
+/obj/item/weapon/storage/proc/dump_it(var/turf/target) //he bought?
+	if(!isturf(target))
+		return
```

Comment:
Shouldn't here be `Adjacent` check?

Reply:
src being set to in view(1) about 12 lines or so above handles it being adjacent, far as I know. tested locally, it's fine, unless some extra janky stuff is happening

Reply:
It's only checks range, but adjacent checks ability to touch to.
For example if there's a windor/window on the way `view(1)` will allow interactions but `adjacent` will not.

Reply:
alright fair enough i'll get something up for that

Reply:
i'm dumb as shit and have no idea how i'd try to slot in an `Adjacent` check

Reply:
The `Adjacent` check seems unnecessary; I've tested it and you can't dump the bag past windoors.

Reply:
> The `Adjacent` check seems unnecessary; I've tested it and you can't dump the bag past windoors.
![Untitled](https://user-images.githubusercontent.com/2458675/65261371-8e574b00-db21-11e9-93ff-d09c434b6a23.png)

Reply:
> have no idea how i'd try to slot in an `Adjacent` check

One way is to pass `user` into `dump_it` and just add ` if(!Adjacent(user)) return`

Reply:
alright so we had different things in mind in relation to the empty contents verb, didn't think that was what was meant. workin on it now